packages/zoe/test/unitTests/contracts/test-otcDesk.js

```diff
@@ -0,0 +1,359 @@
+        Whatever2: moola35Payment,
+      };
+
+      const seat = await zoe.offer(invitation, proposal, payments);
+
+      await t.throwsAsync(() => E(seat).getOfferResult(), {
+        message:
+          'Your offer did not match the quote. See the console log for more details.',
+      });
+
+      await assertPayoutAmount(
+        t,
+        bucksIssuer,
+        E(seat).getPayout('Whatever1'),
+        bucks(500),
+        'bob bucks',
+      );
+      await assertPayoutAmount(
+        t,
+        moolaIssuer,
+        E(seat).getPayout('Whatever2'),
+        moola(35),
+        'bob moola',
+      );
+      await assertPayoutAmount(
+        t,
+        simoleanIssuer,
+        E(seat).getPayout('Whatever3'),
+        simoleans(0),
+        'bob simolean',
+      );
+    },
+  });
+};
+
+const {
+  moolaKit,
+  simoleanKit,
+  bucksKit,
+  moola,
+  simoleans,
+  moolaIssuer,
+  simoleanIssuer,
+  bucksIssuer,
+  bucks,
+  zoe,
+} = setup();
+
+const issuers = {
+  moolaIssuer,
+  simoleanIssuer,
+  bucksIssuer,
+  moola,
+  simoleans,
+  bucks,
+};
+
+const makeInitialPayments = (moolaValue, simoleanValue, bucksValue) => ({
+  moolaPayment: moolaKit.mint.mintPayment(moola(moolaValue)),
+  simoleanPayment: simoleanKit.mint.mintPayment(simoleans(simoleanValue)),
+  bucksPayment: bucksKit.mint.mintPayment(bucks(bucksValue)),
+});
+const timer = buildManualTimer(console.log);
+
+test('zoe - otcDesk', async t => {
+  const installation = await installCode(zoe);
+
+  // Make Alice
+  const alicePayments = makeInitialPayments(10000, 10000, 10000);
+  const alice = await makeAlice(t, zoe, installation, issuers, alicePayments);
+
+  // Make Bob
+  const bobPayments = makeInitialPayments(10000, 10000, 10000);
+  const bob = await makeBob(t, zoe, installation, issuers, bobPayments);
+
+  await alice.startInstance();
+
+  // Alice wants to make a custom quote for Bob. If Bob gives 4
+  // simoleans, he can get 3 moola.
+
+  // First, Alice must add enough inventory. If the contract hasn't
+  // been told of an issuer yet, she must add the issuer to the
+  // contract in the call to make the invitation
+  await alice.addInventory();
+
+  // Alice makes a custom quote for Bob
+  const invitation1 = await alice.makeQuoteForBob(
+    { Simolean: simoleans(4) },
+    { Moola: moola(3) },
+    timer,
+    1,
+  );
+
+  await bob.offerGood(invitation1);
+  await alice.removeInventory();
+
+  // Alice makes a custom quote for Bob
+  const invitation2 = await alice.makeQuoteForBob(
+    { Simolean: simoleans(4) },
+    { Moola: moola(3) },
+    timer,
+    1,
+  );
+  timer.tick();
+
+  // Bob tries to offer but the quote is expired.
+  await bob.offerExpired(invitation2);
+
+  const invitation3 = await alice.makeQuoteForBob(
+    { Buck: bucks(500), Moola: moola(35) },
+    { Simolean: simoleans(15) },
+    timer,
+    100,
+  );
+
+  // Bob tries to offer but he wants more than what was quoted.
+  await bob.offerWantTooMuch(invitation3);
+});
```

Comment:
As it seems to be part fo the design, I'd like to see tests where
* Alice makes a quote for Bob that isn't yet funded. She might or might not fund it before Bob exercises. 
* Alice makes a quote that is currently funded, but removes the funding before Bob receives it.

Reply:
> Alice makes a quote that is currently funded, but removes the funding before Bob receives it.

This will be impossible in the new design, since the funding is escrowed in the covered call.

Reply:
> Alice makes a quote for Bob that isn't yet funded. She might or might not fund it before Bob exercises.

This is also impossible in the new design.